### ISSUE TITLE
Problem in generating sentences with GPT2

### ISSUE THREAD

**Issue by JohnPFL** [huggingface/transformers#7804]
## Environment info

     
- `transformers` version: 3.3.0
- Platform: Linux-3.10.0-1062.12.1.el7.x86_64-x86_64-with-centos-7.7.1908-Core
- Python version: 3.6.8
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: yes

Hello,  
@LysandreJik  @TevenLeScao.
I'm using GPT2 to generate some text, following this guide over here (https://towardsdatascience.com/fine-tuning-gpt2-for-text-generation-using-pytorch-2ee61a4f1ba7). I used the same modification as the article shows in the run_language_modeling.py and run_generation.py. 
I train the model and that works fine, but when I ask the model to generate some texts it just shows me this: 
`"<BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> <BOS> "`
or 
`"<BOS> <PAD><PAD><PAD><PAD><PAD><PAD><PAD><PAD><PAD><PAD>"`
I deeply suspect something's wrong with personalized tokenizer or in the padding process, but I can't really figure out how to solve the problem. 
Probably I'm doing something naive.

A model message I want to show here is: 
Setting `pad_token_id` to 50256 (first `eos_token_id`) to generate sequence
which seems to me a little bit tricky.

Thanks in anticipation.
